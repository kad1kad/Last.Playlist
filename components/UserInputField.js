import { motion } from "framer-motion";

function UserInputField({ handleChange, onSubmit, error }) {
  return (
    // Animates Input on page load
    <motion.div
      initial={{ opacity: 0, y: -60 }}
      animate={{ opacity: 1, y: 10 }}
      transition={{ type: "spring", damping: 20, duration: 0.5 }}
      className=" py-4 flex flex-col justify-center items-center  scroll-smooth"
    >
      {error && (
        <p
          id="userNotFoundError"
          role="alert"
          aria-live="assertive"
          className="text-center text-sm mb-5 text-red-500"
        >
          {error}
        </p>
      )}

      {/* Animates Input moving up after MusicList mounts */}
      <motion.form
        layout
        transition={{ type: "spring", duration: 1.3 }}
        onSubmit={onSubmit}
      >
        <input
          aria-errormessage="userNotFoundError"
          aria-invalid="true"
          type="text"
          className="rounded-full border-2 mr-5 py-4 px-6 text-slate-900 bg-white tracking-wide"
          onChange={handleChange}
          placeholder="Last.fm User"
          spellCheck="false"
        />
        <button
          type="submit"
          className="bg-[#101113] tracking-widest text-neutral-200 px-8 py-4 rounded-full hover:scale-105 transition-all active:scale-95 w-26"
        >
          Submit
        </button>
      </motion.form>
    </motion.div>
  );
}

export default UserInputField;
