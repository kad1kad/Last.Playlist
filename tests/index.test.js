import { render, screen, waitFor } from "@testing-library/react";
import "@testing-library/jest-dom";
import Home from "../pages/index";
import userEvent from "@testing-library/user-event";
import UserInputField from "../components/UserInputField";

jest.mock("next-auth/react", () => {
  const originalModule = jest.requireActual("next-auth/react");
  const mockSession = {
    expires: new Date(Date.now() + 2 * 86400).toISOString(),
  };
  return {
    __esModule: true,
    ...originalModule,
    useSession: jest.fn(() => {
      return { data: mockSession };
    }),
  };
});

describe("Login Component", () => {
  it("Show Sign out when not signed in", async () => {
    render(<Home />);
    const singOutBtn = screen.getByRole("button", { name: /sign out/i });

    expect(singOutBtn).toBeInTheDocument();
  });
});

describe("Header Component", () => {
  test("Displays correct Page Heading and Subtitle", () => {
    render(<Home />);
    const pageHeading = screen.getByRole("heading", { name: "Last.Playlist" });
    const subHeading = screen.getByRole("heading", {
      name: "Last.fm to Spotify Playlist",
    });

    expect(pageHeading).toBeInTheDocument();
    expect(subHeading).toBeInTheDocument();
  });
});

describe("UserInputField", () => {
  test("Render UserInputField", () => {
    render(<Home />);

    const userInput = screen.getByRole("textbox", { id: "userInput" });

    expect(userInput).toBeInTheDocument();
  });

  test("Can type into UserInputField", async () => {
    render(<Home />);

    getUserName();
    userEvent.type(getUserName(), "test");

    await waitFor(() => expect(getUserName().value).toBe("test"));
  });

  test("form is being submited", async () => {
    const onSubmit = jest.fn();
    render(<UserInputField onSubmit={onSubmit} />);
    userEvent.click(screen.getByRole("button", { name: /submit/i }));

    await waitFor(() => {
      expect(onSubmit).toHaveBeenCalled();
    });
  });

  test("errorhandling", async () => {
    const onSubmit = jest.fn();
    render(<UserInputField onSubmit={onSubmit} />);

    getUserName();

    userEvent.type(getUserName(), "thisUserDoesNotExist");

    userEvent.click(screen.getByRole("button", { name: /submit/i }));

    await waitFor(() => {
      expect(screen.getByText(/user not found/i).toBeInTheDocument());
    });
  });
});

function getUserName() {
  return screen.getByRole("textbox");
}
