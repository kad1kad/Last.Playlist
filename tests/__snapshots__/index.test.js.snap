// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component Show Log Out when has session 1`] = `
<div>
  <nav
    class="flex gap-3 text-xs"
  >
    <button
      class="border-neutral-200 text-neutral-200 tracking-wider w-20 h-8 border-2 rounded-full"
    >
      Sign out
    </button>
  </nav>
</div>
`;
